core/lib/Cardano/Script.hs

```diff
@@ -138,8 +184,8 @@ encodeFoldable encode xs = wrapArray len contents
 -- `Api.serialiseToRawBytes $ Api.scriptHash script` realizes
 -- This is basically doing the symbolically the following (using symbols from toCBOR):
 -- digest $ (nativeMultisigTag <> toCBOR multisigScript )
-toScriptHash :: Script -> ByteString
-toScriptHash script = digest $ nativeMultiSigTag <> toCBOR'' script
+toScriptHash :: Script -> ScriptHash
+toScriptHash script = ScriptHash $ digest $ nativeMultiSigTag <> toCBOR'' script
```

Comment:
I am not sure to understand this `nativeMultisigTag` and `toCBOR''` 

Is this to abide by the ad-hoc format used by cardano-cli ? If that's the case, then I think it is wrong. The script hash should be only the blake2b_224 hash of it. We _may_ want to have some compatibility layer with cardano-cli at some point but it shouldn't be the default.

Reply:
so I wanted two-step golden test here, namely for (Api stands for cardano-api):
1. Api.serialiseToCBOR script
2. Api.serialiseToRawBytes $ Api.scriptHash script

But there is problem here how cbor is handled. Symbolically,
    Api.serialiseToCBOR script -> toCBOR [0,multisigScript]
    Api.scriptHash script -> digest $ (nativeMultisigTag <> toCBOR multisigScript )
and this is reflected in the toCBOR' and toCBOR''
This is how cardano-node handles it, spent some time debugging that, and could only adjust to it

Reply:
But cardano-api has its own ad-hoc format for serializing things which is not actually matching the on-chain binary representation (but is a little wrapper around it). I don't think we should bother here and should treat ScriptHash as plain bytes like they are. The fact that cardano-api encodes them as CBOR with a specific tag is superfluous _at this level_. 

Later on, I am all for reconciating both interfaces in one common format, or have ways to convert from the "plain" bytes to cardano-api-specific bytes but arguably, that is something which should live in `cardano-cli`, not in here.

Reply:
as I underlined. I included this as I tried to do it in steps, first replicate toCBOR, then scritpHash calculations. It tutn out later that cardano-node has very surprising impl of scriptHash. Agree that it should be in one place - `adrestia-core` :-) and then they should use it in other repos:-)

Reply:
also please notice that I used this PR (here because of dependiencies) https://github.com/input-output-hk/cardano-transactions/pull/23 to get cbor and script hash values